grails-app/services/org/pih/warehouse/shipping/ShipmentService.groovy

```diff
@@ -2130,4 +2131,45 @@ class ShipmentService {
         return count
     }
 
+
+    void updateOrCreateOrderBasedShipmentItems(Order order, Shipment shipment) {
+        shipment.name = order.name
+        shipment.description = order.description
+        shipment.origin = order.origin
+        shipment.destination = order.destination
+
+        if (order.orderItems) {
+            def itemsToRemove = shipment.shipmentItems.findAll {
+                sItem -> !order.orderItems?.any { oItem -> sItem.orderItems?.any { it.id == oItem.id } }
+            }
+
+            itemsToRemove.each { it -> deleteShipmentItem(it) }
+
+            order.orderItems.each { OrderItem orderItem ->
+                def shipmentItem = shipment.shipmentItems.find { it.orderItems?.any { it.id == orderItem.id } }
+                def lotNumber = shipmentItem?.lotNumber
+                def expirationDate = shipmentItem?.expirationDate
+                if (!shipmentItem) {
+                    shipmentItem = new ShipmentItem(
+                        product: orderItem.product,
+                        inventoryItem: orderItem.inventoryItem,
```

Comment:
@awalkowiak `orderItem.inventoryItem` will probably be null 99% of the time, if not 100%. If there's a product and lot number specified then we need to do a look up of the InventoryItem by product and lot number. But I can't tell from the code what's going on here and whether we need to use InventoryService.findOrCreateInventoryItem() or InventoryService.findInventoryItemByProductAndLotNumber() or something else. 

But all this logic looks complicated so let's have a discuss before we move this one forward. Can you enumerate the different use cases that are represented here:
- User goes back to edit order item quantity
- User goes back to edit order item product (?)
- ...

Reply:
@jmiranda you're right inventory item is not required here. I removed it. Use cases:
- edit line item (qty)
- add new line item
-delete line item

Reply:
@awalkowiak But if there's a lot number and expiration date, it is required. Let's discuss on Slack.